packages/blinks/src/hooks/evm/useEvmWagmiAdapter.ts

```diff
@@ -0,0 +1,104 @@
+import { type ActionAdapter } from '@dialectlabs/blinks-core';
+import type { SignMessageData } from '@solana/actions-spec';
+import { useMemo } from 'react';
+import { deserialize, useAccount, useChains, useConnectorClient } from 'wagmi';
+
+import type { ByteArray, Hex, SignableMessage } from 'viem';
+import {
+  sendTransaction,
+  signMessage,
+  waitForTransactionReceipt,
+} from 'viem/actions';
+
+interface EvmWagmiAdapterParams {
+  onConnectWalletRequest: () => Promise<string | null>;
+}
+
+export function useEvmWagmiAdapter({
+  onConnectWalletRequest,
+}: EvmWagmiAdapterParams) {
+  const chains = useChains();
+  const { address } = useAccount();
+  const { data: client } = useConnectorClient();
+
+  const adapter = useMemo(
+    () =>
+      ({
+        metadata: {
+          supportedBlockchainIds: chains.map((chain) => `eip155:${chain.id}`),
+        },
+        connect: async () => {
+          if (!address) {
+            try {
+              const potentialAddress = await onConnectWalletRequest();
+              return potentialAddress || null;
+            } catch (error) {
+              console.error('Blink wagmi adapter connection failed:', error);
```

Comment:
Do we need this log here? I would assume that it would throw if user cancelled connect for instance

Reply:
I thought to keep it for a time being while new devs will be integrating it. should be easier for us to help & debug just in case